test/e2e/daemonset/daemonset.go

```diff
@@ -0,0 +1,34 @@
+package daemonset
+
+import (
+	"context"
+	"time"
+
+	"github.com/onsi/gomega"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func Eventually(daemonSetKey types.NamespacedName) gomega.AsyncAssertion {
+	return gomega.Eventually(func() appsv1.DaemonSet {
+		daemonSet := appsv1.DaemonSet{}
+		_ = framework.Global.Client.Get(context.TODO(), daemonSetKey, &daemonSet)
```

Comment:
We can use `GetDaemonSet` here.

About error handling, if we make the function return daemonset and error Eventually will fail if there is an error, from gomega doc https://onsi.github.io/gomega/#eventually:

> The function that you pass to Eventually can have more than one return value. In that case, Eventually passes the first return value to the matcher and asserts that all other return values are nil or zero-valued. This allows you to use Eventually with functions that return a value and an error – a common pattern in Go

Reply:
Like the thing about error. Done.